olive/passes/onnx/optimum_conversion.py

```diff
@@ -41,22 +60,25 @@ def _run_for_config(
         # TODO(jambayk): export into temp dir and then move to sub-dirs of output_model_path
         # so that we only keep the final model files in the output_model_path
         # and track external data if present
+        config["extra_args"] = config["extra_args"] or {}
+        config["extra_args"].update(
+            {
+                "opset": config["target_opset"],
+                "fp16": config["fp16"],
+                "no_post_process": config["no_post_process"],
+                "device": config["device"],
+            }
+        )
         hf_config = deepcopy(model.hf_config) or HFConfig()
-        if version.parse(optimum_version.__version__) < version.parse("1.14.0"):
-            export_optimum_model(
-                model.model_path or hf_config.model_name,
-                output_model_path,
-                opset=config["target_opset"],
-                no_post_process=True,
-            )
-        else:
-            export_optimum_model(
-                model.model_path or hf_config.model_name,
-                output_model_path,
-                opset=config["target_opset"],
-                legacy=True,
-                no_post_process=True,
-            )
+        if version.parse(optimum_version.__version__) >= version.parse("1.14.0"):
+            # Optimum 1.14.0 needs to be run in legacy mode to support older versions of transformers
+            config["extra_args"]["legacy"] = True
```

Comment:
do you have plans to deprecate legacy support or make it default to `False`. Since optimum is planning to deprecate legacy mode, I think we should have legacy as opt-in instead of the default.

Reply:
Yes, we should deprecate the legacy way by default and provide the options for users to enable for some corner cases in the future. 
I remembered that when I test the way with legacy=False for optimum 1.14.0, the open llama case failed for some reasons. 

Added the TODO item, will try to investigate the failure and remove legacy once the validations pass.

Reply:
Thanks! this sounds good to me.